vscode-dotnet-sdk-extension/src/extension.ts

```diff
@@ -157,19 +157,24 @@ export function activate(context: vscode.ExtensionContext, extensionContext?: IE
 }
 
 function setPathEnvVar(pathAddition: string, displayWorker: IWindowDisplayWorker) {
-    if (process.env.PATH && process.env.PATH.includes(pathAddition)) {
-        // No need to add to PATH again
-        return;
-    }
-
     let pathCommand: string;
     if (os.platform() === 'win32') {
+        if (process.env.PATH && process.env.PATH.includes(pathAddition)) {
+            // No need to add to PATH again
+            return;
+        }
         pathCommand = `for /F "skip=2 tokens=1,2*" %A in ('%SystemRoot%\\System32\\reg.exe query "HKCU\\Environment" /v "Path" 2^>nul') do ` +
-                      `(%SystemRoot%\\System32\\reg.exe ADD "HKCU\\Environment" /v Path /t REG_SZ /f /d "${pathAddition};%C")`;
+            `(%SystemRoot%\\System32\\reg.exe ADD "HKCU\\Environment" /v Path /t REG_SZ /f /d "${pathAddition};%C")`;
```

Comment:
we should make sure it writes to user level Environment. https://github.com/dotnet/sdk/blob/b91b88aec2684e3d2988df8d838d3aa3c6240a35/src/Cli/Microsoft.DotNet.Cli.Utils/WindowsRegistryEnvironmentPathEditor.cs#L46

If vscode is elevated. It will cause a lot of problem.

And on ${pathAddition};%C. You should make sure it is not expand. So if there is a path

%USERPROFILE%\Other

Reply:
Just verified that this writes to the same reg key as in the like, HKCU (the key we're using here) is shorthand for HKEY_CURRENT_USER, where the SDK code writes. 

> it should still be %USERPROFILE%\Other;newpath, instead of c:\user\myname\Other;newpath

Just verified that this is true with the current implementation.